data/filters/youtube-channel.yaml

```
@@ -0,0 +1,29 @@
+title: Hide Youtube videos by channel
+params:
+  - name: channels
+    description: Channels to filter out
+    type: list
+    default: [ "ChannelURL" ]
+tags:
+  - youtube
+template: |
+  {{#each channels}}
+  www.youtube.com##ytd-browse[page-subtype="home"] ytd-rich-item-renderer:has([href*="{{ . }}"])
+  www.youtube.com##ytd-search ytd-video-renderer:has([href*="{{ . }}"])
+  www.youtube.com##ytd-search ytd-channel-renderer:has([href*="{{ . }}"])
+  www.youtube.com##ytd-search #channel-name:has-text({{ . }}):upward(ytd-secondary-search-container-renderer)
+  www.youtube.com##ytd-watch-next-secondary-results-renderer #channel-name:has-text({{ . }}):upward(ytd-compact-video-renderer)
+  {{/each}}
+tests:
+  - params: {}
+    output: ""
+  - params:
+      keywords: [ "ChannelURL" ]
```

To fix the tests after the param rename
```suggestion
      channels: [ "ChannelURL" ]
```